ykcompile/src/lib.rs

```diff
@@ -222,6 +234,63 @@ impl TraceCompiler {
         Ok(())
     }
 
+    /// Compile a call to a native symbol. This is used for occasions where you don't want the
+    /// callee inlined, or cannot inline it (e.g. it's a foreign function).
+    fn c_call(
+        &mut self,
+        opnd: &CallOperand,
+        args: &Vec<Operand>,
+        dest: &Option<Place>,
+    ) -> Result<(), CompileError> {
+        let sym = if let CallOperand::Fn(sym) = opnd {
+            sym
+        } else {
+            return Err(CompileError::Unimplemented(
+                "unknown call target".to_owned(),
+            ));
+        };
+
+        // FIXME implement support for arguments.
+        if !args.is_empty() {
+            return Err(CompileError::Unimplemented("call with args".to_owned()));
+        }
+
+        // Generally speaking, we don't know what the callee will clobber, so we save our registers
+        // before doing the call. For now we assume that the caller is OK with RAX being clobbered
+        // though.
+        for reg in 1..X86_86_NUM_REGS {
+            dbg!(reg);
```

Comment:
I think we should remove these `dbg!`'s before merging. There's another one below.

Reply:
Oops. That was unintentionally left-in.